### PR TITLE
Add super editor tests

### PR DESCRIPTION
**Pull request by matthew-carroll**
This PR adds the `super_editor` mono repo to the test registry.

`super_editor` is here:
https://github.com/superlistapp/super_editor

I wanted to define all the tests in a single script, but I couldn't get it to work:

- First, I tried placing all test commands in a single file. When I ran that locally, it hung forever without any output. When I tried that in CI, it failed for some reason.
- I gave up and divided that file into a different script for every project in the repo. That failed because Windows didn't recognize the script file without an extension. So then I was forced to separate `.sh` from `.bat` and configure per-platform.

We now have a regrettable number of scripts in our super_editor repo to integrate with this test repo. I'd like to reduce these scripts down to one, if possible.

### CODE REVIEW

**Comment by matthew-carroll** [comment]
@Hixie should I tag someone else for the review of this PR?

File: registry/super_editor.test
Comment (Hixie) [inline registry/super_editor.test:23]: throughout this file you can remove the `./` and `.\` prefixes

Comment (Hixie) [reply registry/super_editor.test:23]: (i think? if not please let me know, it would be weird if they were required but i guess it's possible)

Comment (matthew-carroll) [reply registry/super_editor.test]: Ok, done. We'll see if CI accepts it.

File: registry/super_editor.test
Comment (Hixie) [inline registry/super_editor.test:23]: my bad, i didn't notice these weren't in a subdirectory. for things in the root of the repo you do need `./`.

To be precise, the requirement is that this be a path, so it must have at least one `/` (or `\` for windows), so that it doesn't get affected by the $PATH env variable. So when you have a directory (as in the `update` lines above) you don't need `./`, but when you only have a script name, you do. i should document this better, sorry.

File: registry/super_editor.test
Comment (Hixie) [inline registry/super_editor.test:38]: you may remove this comment also if you like (up to you, if you want it for your own purposes feel free to keep it)

Comment (matthew-carroll) [reply registry/super_editor.test:38]: I'll leave it in to remind me

**Comment by matthew-carroll** [comment]
@Hixie do you know if it's possible to reduce these scripts down to a single file, or maybe just one for Windows and one for Posix? Right now we've got 6 of them sitting in our top-level directory: 

https://github.com/superlistapp/super_editor

**Comment by Hixie** [comment]
In principle I would expect you can just have one script that calls the other three; does that work? Having several scripts is fine from the perspective of this repo, fwiw.

Typically these kinds of scripts are put in a subdirectory like `ci` or `dev` along with other scripts and tools for supporting the tests.

File: registry/super_editor.test
Comment (Hixie) [inline registry/super_editor.test:33]: Did you try running them locally as per this comment? I tried running the tests locally but they seem to take a long time to run (they haven't finished yet, and I'm pretty sure we're beyond 15 minutes).

Comment (matthew-carroll) [reply registry/super_editor.test:33]: When I ran locally they seemed to hang forever. But even in CI they do take a long time. I'm not sure why. Do you have any recommendations for figuring out what's taking so long and how to reduce the time? I don't think we're running any golden tests, at least that I recall. And the vast majority of tests are either Dart tests or pretty short widget tests...

Comment (Hixie) [reply registry/super_editor.test:33]: Ok, they took 30 real-time minutes to run 100 times on my machine (163 CPU minutes).

Let me take a quick look at the tests to see if there's anything obvious there. In general I would expect slow tests to mean either that there's a lot of tests, that the tests do a lot of looping, or that the tests wait real time rather than relying on the async systems.

Comment (Hixie) [reply registry/super_editor.test:33]: It's just that you have a lot of tests. I scanned through them and they seem very valuable and worth having despite the cost, so don't worry about it. I didn't see any red flags.

Comment (matthew-carroll) [reply registry/super_editor.test:33]: Great. Thanks. I'm going to try to relocate and possibly de-dup these scripts before merging. So please expect that after approval, or you can hold approval until then, if you want to.

**Comment by Hixie** [comment]
FWIW, merging the three scripts into one worked for me.

**Comment by matthew-carroll** [comment]
> FWIW, merging the three scripts into one worked for me.

Was that with a `cd /path` and `cd ../` for each project?

**Comment by Hixie** [comment]
I just had one test file like this:
```bash
(cd attributed_text; flutter test)
(cd super_editor; flutter test)
(cd super_text_layout; flutter test)
```

...and when I ran it the output was:
```
00:01 +0: loading /home/ianh/dev/flutter/dev/customer_testing/super_editor/attributed_text/test/attributed_spans_test.dart                       
Initializing logger: attributions
Deactivating logger: attributions
00:01 +49: All tests passed!                                                                                                                     
00:12 +730 ~4: All tests passed!                                                                                                                 
00:03 +28: All tests passed!                                                                                                                     
```

FWIW, looks like there's a few spurious `print` calls in the tests, you can enable `avoid_print` in your analysis options to catch those.

**Comment by Hixie** [comment]
(I didn't test this on Windows, but I imagine something similar should work there.)

**Comment by matthew-carroll** [comment]
Thanks. I'll give it a try.

**Comment by matthew-carroll** [comment]
BTW @Hixie do yo know if we need `.bat` and `.sh` files? Or should it be possible to have a single file without an extension that's run everywhere? The commands that you're sending don't seem to have any platform-specific notation...

**Comment by Hixie** [comment]
If you can find a way to write a polyglot file that's valid both as a windows batch file and a shell script, then you should be able to use a single file to run on both. In general I find writing batch files and shell scripts is hard enough without the extra complexity of making them polyglot. :-)

Another option is to teach `flutter test` to determine which project the given directory is actually in rather than using the current directory, then you could just say `test=flutter test super_text_layout/test` and so on and not have a shell script at all. cc @christopherfujino

**Comment by matthew-carroll** [comment]
@Hixie do the current CI failure messages make sense to you?

**Comment by Hixie** [comment]
> @Hixie do the current CI failure messages make sense to you?

The script is run from the root of the repo, so when the script tries to `cd ../attributed_text` it's trying to escape the repo.

**Comment by matthew-carroll** [comment]
@Hixie I can't seem to find a path that works. I've tried:

- "../attributed_text"
- "./attributed_text"
- "attributed_text"

Do you know how these paths should be specified?

**Comment by matthew-carroll** [comment]
@Hixie I got Windows to work. But now Linux is timing out....

**Comment by matthew-carroll** [comment]
@Hixie checking in again. I'd love to this ironed out and merged in as soon as possible. I'm worried that I'm going to forget this PR is open, if we don't merge it pretty soon.

**Comment by Hixie** [comment]
Looks like the tests are just taking a long time; it more or less doubled the total time of the shard and that's hitting the one hour quota on Linux (and getting very close to hitting the quotas on mac and windows).

**Comment by Hixie** [comment]
Submitted https://github.com/flutter/tests/pull/140 to shard the tests, which should let these finish.

**Comment by Hixie** [comment]
In general we're trying to minimize the number of timeouts because they just end up making things more flaky.

**Comment by Hixie** [comment]
ok try rebasing the PR and repushing it

**Comment by matthew-carroll** [comment]
@Hixie it looks like we're all green. Thanks for the hand-holding.

**Comment by matthew-carroll** [comment]
Also, to be clear, I am still waiting on an approval so that I can merge :)

**Review by Hixie — APPROVED** [review]
LGTM

**Pull request merged**
